src/v/storage/disk_log_impl.cc

```diff
@@ -572,15 +572,32 @@ offset_stats disk_log_impl::offsets() const {
     // with reverse iterators, so we manually find the iterator
     segment_set::type end;
     segment_set::type term_start;
+    segment_set::type last_committed;
     for (int i = (int)_segs.size() - 1; i >= 0; --i) {
         auto& seg = _segs[i];
         if (!seg->empty()) {
             if (!end) {
                 end = seg;
             }
+
+            /**
+             * move back until we find a segment with initialized committed
+             * offset
+             */
+            if (
+              !last_committed
+              || last_committed->offsets().committed_offset
+                   < model::offset{0}) {
+                last_committed = seg;
+            }
+            if (
+              last_committed->offsets().committed_offset
+              && seg->offsets().term < end->offsets().term) {
```

Comment:
can you update the comment above to include the rationale behind the term conditional here? the comment only mentions the offset condition. I'm assuming its because when the segment has an out of range term none of its content should be considered?

Reply:
I've added comment to explain what is happening in the loop .